app/src/main/kotlin/compiler/intermediate/ControlFlowPlanner.kt

```diff
@@ -485,15 +484,6 @@ class ControlFlowPlanner(private val diagnostics: Diagnostics) {
                 is AssignmentTarget.VariableTarget -> {
                     val owner = variableProperties[Ref(target.variable)]!!.owner
                     val accessGenerator = variableAccessGenerators[Ref(owner)]!!
-
-                    target.variable.type.takeIf { it is Type.Array }?.let {
-                        cfgBuilder.addNextCFG(
-                            arrayMemoryManagement.genRefCountDecrement(
-                                accessGenerator.genRead(target.variable, owner === currentFunction),
-                                it
-                            )
-                        )
-                    }
```

Comment:
Why is it removed? Is that case handled somewhere else?

Reply:
I think we should first decrease (because the returning expression has just completed) and then increase (because the target variable is a new reference). I gathered from debugging that the increase is missing. To fix that, I just removed a decrease instead of adding the missing increase.

Reply:
Nevermind, I was looking at the wrong bit of code

Reply:
I didn't notice this removal is not necessary at all. Good catch :smile: